src/algebra/algebra/spectrum.lean

```diff
@@ -68,6 +68,14 @@ variable {R}
 noncomputable def resolvent (a : A) (r : R) : A :=
 ring.inverse (algebra_map R A r - a)
 
+/-- The unit `1 - r⁻¹ • a` constructed from `r • 1 - a` when the latter is a unit. -/
+@[simps]
+noncomputable def is_unit.sub_inv_smul {r : Rˣ} {s : R} {a : A}
```

Comment:
I now notice `is_unit.smul_sub_iff_sub_inv_smul` just below.

Can we can unify that lemma and this def. For example, the following is possible:
```lean
@[simps]
noncomputable def is_unit.sub_inv_smul (R A : Type*) [comm_ring R] [ring A] [algebra R A]
  {r : Rˣ} {a : A}
  (h : is_unit $ r • 1  - a) : Aˣ :=
(is_unit.smul_sub_iff_sub_inv_smul.mp h).unit
```
though I appreciate this is a slightly different, weaker statement.

At the very least I think we should share `variables` between these two.

Reply:
variables shared, but I actually need the stronger statement. I need to know how the inverse of the new unit is related to the inverse of the old unit. This is for `units_smul_resolvent` below.

Reply:
I also just simplified the proof of `is_unit.smul_sub_iff_sub_inv_smul`